AdvPromptSet/utils.py

```diff
@@ -0,0 +1,187 @@
+                  'homosexual_gay_or_lesbian', 'bisexual', 'other_sexual_orientation',
+                  'christian', 'jewish', 'muslim', 'hindu', 'buddhist', 'atheist',
+                  'other_religion', 'black', 'white', 'asian', 'latino',
+                  'other_race_or_ethnicity', 'physical_disability',
+                  'intellectual_or_learning_disability',
+                  'psychiatric_or_mental_illness', 'other_disability']
+
+    df_test_public = pd.read_csv(os.path.join(dat_folder, df_name, "test_public_expanded.csv"))
+    df_test_private = pd.read_csv(os.path.join(dat_folder, df_name, "test_private_expanded.csv"))
+
+    df_train = df_train[['id', 'comment_text']+toxic_cols+group_cols]
+    df_test_public = df_test_public[['id', 'comment_text']+toxic_cols+group_cols]
+    df_test_private = df_test_private[['id', 'comment_text']+toxic_cols+group_cols]
+
+    df_test = pd.concat([df_test_public, df_test_private], axis=0).reset_index(drop=True).copy()
+
+    df_test = df_test.assign(test_set=1)
+    df_train = df_train.assign(test_set=0)
+
+    df_jigsaw = pd.concat([df_test, df_train], axis=0).reset_index(drop=True).copy()
+
+    # look at rows with sensitive group labels
+    df_has_group = df_jigsaw.dropna(axis=0, subset=group_cols, how="all").reset_index(drop=True).copy()
+
+    # look at rows with/o sensitive group labels (only toxicity labels)
+    df_toxic_only = df_jigsaw[df_jigsaw.isnull().any(axis=1)]
+    df_toxic_only = df_toxic_only.drop(group_cols, axis=1).reset_index(drop=True).copy()
+
+    assert df_toxic_only.shape[0] + df_has_group.shape[0] == df_jigsaw.shape[0]
+    assert df_toxic_only.isna().sum().sum() == 0  # no missing in df_toxic_only
+    assert df_has_group.isna().sum().sum() == 0  # no missing in df_has_group
+
+    def binarize(x):
+        return np.where(x >= 0.5, 1, 0)
+
+    # axis=0 means apply to columns; axis=1 to rows
+    df_toxic_only[toxic_cols] = df_toxic_only[toxic_cols].apply(binarize, axis=0)
+    df_has_group[toxic_cols+group_cols] = df_has_group[toxic_cols+group_cols].apply(binarize, axis=0)
+
+    df_toxic_only = df_toxic_only.assign(jigsaw_dat=2)
+    df_has_group = df_has_group.assign(jigsaw_dat=2)
+
+    df_toxic_only = df_toxic_only.assign(identity_grp="predicted")
+    df_has_group = df_has_group.assign(identity_grp="annotated")
+
+    return df_toxic_only, df_has_group
+
+
+def process_jigsaw_all(df_jigsaw1, df_toxic_only, df_has_group):
+    df_jigsaw_lst = [df_jigsaw1, df_toxic_only, df_has_group]
+
+    df_lst_processed = list()
+    for df in df_jigsaw_lst:
+        df = remove_long_rows(df)
+        df = augment_row(df)
+        df = df[["id", "comment_text", "toxicity", "jigsaw_dat", "test_set", "identity_grp"]].copy()
+        df_lst_processed.append(df)
+
+    df_jigsaw = pd.concat(df_lst_processed, axis=0).reset_index(drop=True).copy()
+    return df_jigsaw
+
+
+def check_shape(term, df):
```

Comment:
Hmm - what is the output supposed to be here? One idea could be to add an `assert` to make sure it's the correct shape/counts, up to you

Reply:
Removed. Thanks for catching. Those are scripts for debugging purposes.